### PR TITLE
Print fatal errors

### LINKED ISSUE #2961: `IO#unsafeRunAsync` and `unsafeRunAndForget` eat fatal errors
What happens here is `unsafeRunFiber` sets up the fatal error callback relocation thing, but then `unsafeRunAsync` has to box the error into a `Left`. This *may* work, but even if it works, `unsafeRunAndForget` is just going to drop it on the floor and the `IORuntime` silently dies. That seems like a bad experience.

I think what should probably happen here is we should register the callback as usual, but then in these kinds of cases where can't reliably continue the error relocation, we should just give up and dump the exception to stderr. Note that `unsafeToFuture` doesn't have this problem because the `Promise` is already allocated, so the fatal errors *can* be safely relocated into the resultant `Future` (and processed upon) without running into the cases where the JVM no longer functions.

### PR DESCRIPTION
**Pull request by sbly**
This is meant to address #2961. This is my first PR. I'm not sure I understood the ticket correctly. I'm not sure if this is the right place to address the issue, or if it needs to be done deeper, inside the IO runloop. Could I get some feedback? Thank you

I wrote a test for this locally to observe the behavior, but it seems with or without the change fatal errors get printed, which I think might be because of the test framework though. So I'm not sure how to test this behavior

### CODE REVIEW

File: core/shared/src/main/scala/cats/effect/IO.scala [hunk 1]
```diff
@@ -832,7 +832,11 @@ sealed abstract class IO[+A] private () extends IOPlatform[A] {
    * should never be totally silent.
    */
   def unsafeRunAndForget()(implicit runtime: unsafe.IORuntime): Unit =
-    unsafeRunAsync(_ => ())
+    unsafeRunAsync {
+      case Left(NonFatal(_)) => ()
+      case Left(e) => System.err.println(e)
```
Comment (armanbilge) [inline core/shared/src/main/scala/cats/effect/IO.scala]: I wonder if we can use `e.printStackTrace()` here instead :)

**Review by armanbilge — COMMENTED** [review]
Thanks for the PR! To appease CI you can run `scalafixAll` in sbt. Perhaps we forgot to add this to the `prePR` command, sorry about that :)

File: core/shared/src/main/scala/cats/effect/IO.scala (hunk 1 above)
Comment (sbly) [reply core/shared/src/main/scala/cats/effect/IO.scala]: Whoops, good point :)

**Comment by sbly** [comment]
@armanbilge Thanks for the review! Running `scalafixAll` worked. I checked the `prePR` command and it should be calling it:

```
addCommandAlias(
  "prePR",
  "; root/clean; root/scalafixAll; scalafmtSbt; +root/scalafmtAll; +root/headerCreate")
```

but when I use `prePR` directly, it doesn't seem to call scalafix. I think maybe this has to do with project aggregation in sbt, but I'm not familiar enough with this project's structure to diagnose the issue

**Review by djspiewak — CHANGES_REQUESTED** [review]
Right idea, but not quite there! :-) So the problem with `unsafeRunAndForget` comes from the fact that it is delegating to `unsafeRunAsync`, and `unsafeRunAsync` boxes fatal errors. Note that `unsafeRunAsyncOutcome` has the same issue.

There are two things we could do here, and I think I've convinced myself that one of them is better. Specifically, in `unsafeRunAsync` and `unsafeRunAsyncOutcome`, we check to see if the error is *not* `NonFatal` and then specifically print it in that case. We can still *try* to box it and send it along, but this is basically the last moment that we can be absolutely certain we have our hands on the error before more bad things become possible due to the type signatures.

In theory, `unsafeRunAndForget` shouldn't need to be changed.

**Comment by sbly** [comment]
@djspiewak Thanks for the feedback! I implemented your suggestion.

File: core/shared/src/main/scala/cats/effect/IO.scala [hunk 2]
```diff
@@ -807,16 +809,27 @@ sealed abstract class IO[+A] private () extends IOPlatform[A] {
       implicit runtime: unsafe.IORuntime): Unit = {
     unsafeRunFiber(
       cb(Left(new CancellationException("The fiber was canceled"))),
-      t => cb(Left(t)),
-      a => cb(Right(a)))
+      t => {
+        if (!NonFatal(t)) {
```
Comment (djspiewak) [inline core/shared/src/main/scala/cats/effect/IO.scala:813]: TIL `NonFatal` has an `apply` method.

**Review by djspiewak — APPROVED** [review]
Sorry this took me so long!

**Pull request merged**